content/posts/developing-versionless-graphql-apis.md

```diff
@@ -0,0 +1,107 @@
+--- 
+title: "Developing Versionless APIs with GraphQL" 
+date: "2020-07-22" 
+authors:  
+- github: xo426017 
+  name: "Bing Xiao" 
+categories: ["graphql", "API"] 
+featuredImage: "/images/developing-versionless-graphql-apis/featured.png" 
+---
+
+GraphQL has recently made a huge splash in the API world by addressing some of the pain points with REST APIs.
+
+With REST APIs, every resource needs an API endpoint. When we need to add a new resource, the endpoint often must be versioned, and/or a brand-new endpoint must be created. This leads to many maintainability and usability issues.
+
+However, GraphQL operates from a single `/graphql` endpoint. The new resource can be added via new types, or fields, in the GraphQL schema without creating a new endpoint.
+
+The GraphQL schema also provides interactive documentation. API consumers can use [GraphiQL](https://www.electronjs.org/apps/graphiql) or the [GraphQL Playground](https://github.com/prisma-labs/graphql-playground) to view the documentation for the query that they are building.
+
+That said, designing an extensible schema is the most important part of building versionless APIs in GraphQL.  
+
+## Schema design
+
+There are a few things that can help when designing a schema.
+
+First and foremost, grasp the domain knowledge, and try to understand the business domain as best you can. Then, follow a few schema design principles for GraphQL. For instance, it is preferable to use more complex structures like an object type than simple structures, like an array. This ensures that new data fields can be added easily, and more complex business objects can be composed without much effort.
```

Comment:
There are two ideas in this paragraph.  The first sentence about understanding the business domain should be expanded with at least one more sentence.  The second sentence should be another paragraph: "Second, follow a few schema..."